gopowerstore_types_test.go

```diff
@@ -75,3 +75,32 @@ func TestAPIError_VolumeAlreadyRemovedFromVolumeGroup(t *testing.T) {
 	apiError.Message = "One or more volumes to be removed are not part of the volume group"
 	assert.True(t, apiError.VolumeAlreadyRemovedFromVolumeGroup())
 }
+
+func TestAPIError_FSCreationLimitReached(t *testing.T) {
```

Comment:
can we re-initialize apiError for each test case to ensure that previous test cases do not affect subsequent ones:
eg:
func TestAPIError_FSCreationLimitReached(t *testing.T) {
    // Case 1: Wrong status code
    apiError := NewAPIError()
    apiError.StatusCode = http.StatusBadRequest
    apiError.Message = "The limit of 125 file systems for the NAS server has been reached"
    assert.False(t, apiError.FSCreationLimitReached())

    // Case 2: Correct status, but unrelated message
    apiError = NewAPIError()
    apiError.StatusCode = http.StatusUnprocessableEntity
    apiError.Message = "some unrelated error message"
    assert.False(t, apiError.FSCreationLimitReached())
//case 3
//case4
...
...
  
}